<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Window</class>
 <widget class="QWidget" name="Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Window</string>
  </property>
  <widget class="QStackedWidget" name="stackedWidget">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>801</width>
     <height>601</height>
    </rect>
   </property>
   <widget class="QWidget" name="menuPage">
    <property name="styleSheet">
     <string notr="true"/>
    </property>
    <widget class="QFrame" name="menuFrame">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>801</width>
       <height>601</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
    background-color: rgb(68, 68, 68);    
}

QLabel#gameTitleLabel{
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QPushButton:!hover {
	color: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}

QPushButton:hover {
    color: black;
	background: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>230</y>
        <width>311</width>
        <height>161</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="challengeButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="mouseTracking">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>Challenge</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>80</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="freePractiseButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Free practise</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="gameTitleLabel">
      <property name="geometry">
       <rect>
        <x>150</x>
        <y>60</y>
        <width>521</width>
        <height>71</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="text">
       <string>SpeedTypingImprover</string>
      </property>
      <property name="textFormat">
       <enum>Qt::PlainText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="languageChoicePage">
    <widget class="QFrame" name="languageChoiceFrame">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>801</width>
       <height>601</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
    background-color: rgb(68, 68, 68);    
}

QLabel#gameTitleLabel{
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QPushButton:!hover {
	color: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}

QPushButton:hover {
    color: black;
	background: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QWidget" name="verticalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>250</x>
        <y>230</y>
        <width>311</width>
        <height>161</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="englishButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="mouseTracking">
          <bool>false</bool>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>English</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="polishButton">
         <property name="text">
          <string>Polski</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_3">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="backButton">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
           <horstretch>1</horstretch>
           <verstretch>1</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Back</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
   <widget class="QGameWidget" name="gamePage">
    <widget class="QFrame" name="gameFrame">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>801</width>
       <height>601</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
    background-color: rgb(68, 68, 68);    
}

QLabel#speedLabel{
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QLabel#speedValueLabel{
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QPushButton:!hover {
	color: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    padding: 6px;
}

QPushButton:hover {
    color: black;
	background: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    padding: 6px;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <property name="lineWidth">
      <number>10</number>
     </property>
     <widget class="QFrame" name="scoreFrame">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>450</y>
        <width>721</width>
        <height>121</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <property name="lineWidth">
       <number>5</number>
      </property>
      <widget class="QLabel" name="netWPMLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>10</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color: white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>NetWPM</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="netWPMValueLabel">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>60</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="errorsLabel">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>10</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>Errors</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="errorsValueLabel">
       <property name="geometry">
        <rect>
         <x>150</x>
         <y>60</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="livesLabel">
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>10</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>Lives</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="livesValueLabel">
       <property name="geometry">
        <rect>
         <x>290</x>
         <y>60</y>
         <width>141</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>14</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>0</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="inputLabel">
       <property name="geometry">
        <rect>
         <x>430</x>
         <y>10</y>
         <width>281</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
      <widget class="QLabel" name="timeLabel">
       <property name="geometry">
        <rect>
         <x>430</x>
         <y>60</y>
         <width>281</width>
         <height>51</height>
        </rect>
       </property>
       <property name="font">
        <font>
         <pointsize>16</pointsize>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">color:white</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::Panel</enum>
       </property>
       <property name="lineWidth">
        <number>2</number>
       </property>
       <property name="text">
        <string>Time: 0</string>
       </property>
       <property name="scaledContents">
        <bool>false</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
      </widget>
     </widget>
     <widget class="QGraphicsView" name="graphicsView">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>70</y>
        <width>721</width>
        <height>371</height>
       </rect>
      </property>
      <property name="frameShape">
       <enum>QFrame::Box</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Plain</enum>
      </property>
      <property name="lineWidth">
       <number>4</number>
      </property>
     </widget>
     <widget class="QPushButton" name="slowerButton">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>30</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Slower</string>
      </property>
     </widget>
     <widget class="QPushButton" name="fasterButton">
      <property name="geometry">
       <rect>
        <x>500</x>
        <y>30</y>
        <width>111</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Faster</string>
      </property>
     </widget>
     <widget class="QPushButton" name="quitButton">
      <property name="geometry">
       <rect>
        <x>700</x>
        <y>30</y>
        <width>61</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Quit</string>
      </property>
     </widget>
     <widget class="QLabel" name="speedLabel">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>20</y>
        <width>151</width>
        <height>41</height>
       </rect>
      </property>
      <property name="text">
       <string>Speed: 1</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QWidget" name="gameOverPage">
    <widget class="QFrame" name="gameOverFrame">
     <property name="geometry">
      <rect>
       <x>0</x>
       <y>0</y>
       <width>801</width>
       <height>601</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">QFrame {
    background-color: rgb(68, 68, 68);    
}

QLabel {
    font-size: 20px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QLabel#gameOverLabel {
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QPushButton:!hover {
	color: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}

QPushButton:hover {
    color: black;
	background: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QLabel" name="gameOverLabel">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>60</y>
        <width>511</width>
        <height>71</height>
       </rect>
      </property>
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">QFrame {
    background-color: rgb(68, 68, 68);    
}

QLabell{
    font-size: 35px;
    color: white;
    font-weight: normal;
    font-style: normal;
    font-variant: small-caps;
    text-transform: none;
}

QPushButton:!hover {
	color: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}

QPushButton:hover {
    color: black;
	background: white;
    border-style: outset;
    border-width: 1px;
    border-radius: 1px;
    border-color: white;
    font: bold 14px;
    min-width: 10em;
    padding: 6px;
}</string>
      </property>
      <property name="text">
       <string>Game Over</string>
      </property>
      <property name="textFormat">
       <enum>Qt::PlainText</enum>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="endBackButton">
      <property name="geometry">
       <rect>
        <x>240</x>
        <y>460</y>
        <width>309</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Back</string>
      </property>
     </widget>
     <widget class="QWidget" name="verticalLayoutWidget_3">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>170</y>
        <width>371</width>
        <height>251</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <widget class="QLabel" name="endNetWPMLabel">
         <property name="text">
          <string>Net WPM:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="endErrorsLabel">
         <property name="text">
          <string>Errors:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="endTimeLabel">
         <property name="text">
          <string>Time:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="endTypedCharactersLabel">
         <property name="text">
          <string>Typed characters:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignCenter</set>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </widget>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>QGameWidget</class>
   <extends>QWidget</extends>
   <header>qgamewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
